compiler/rustc_lexer/src/lib.rs

```diff
@@ -804,8 +812,6 @@ impl Cursor<'_> {
     /// guarded string is not found. It is the caller's
     /// responsibility to do so.
     pub fn guarded_double_quoted_string(&mut self) -> Option<GuardedStr> {
-        debug_assert!(self.prev() != '#');
-
```

Comment:
Should this be replaced with a check of `str_before` in `maybe_report_guarded_str`?

Reply:
I don't think this check should exist at all, and I'm surprised it hasn't blown up yet. It will blow up on `####` if running in debug.